### PR TITLE
Cleanup job: Remove orhpaned files in file system

### PR DESCRIPTION
**Pull request by boyan-velinov**
Jira: LMCROSSITXSADEPLOY-863

### CODE REVIEW

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/jobs/CleanUpJob.java [hunk 1]
```diff
@@ -136,21 +136,38 @@ private void cleanUpFinishedOperationsData(Date expirationTime) {
...
+        LOGGER.info("Deleting old orphaned MTA files modified before: " + expirationTime.toString());
+        try {
+            List<FileEntry> oldFileEntries = fileService.listByModificationTime(expirationTime);
+            List<FileEntry> orhpanedFileEntries = new ArrayList<FileEntry>();
+            List<String> appArchiveIds = activitiFacade.getHistoricVariableInstancesByVariableName(Constants.PARAM_APP_ARCHIVE_ID)
+                .stream()
+                .map(appArchiveId -> (String) appArchiveId.getValue())
+                .collect(Collectors.toList());
+            
+            for (FileEntry oldFileEntry : oldFileEntries) {
```
Comment (enchobelezirev) [inline com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/jobs/CleanUpJob.java]: Stream::filter?

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/jobs/CleanUpJob.java [hunk 2]
```diff
@@ -136,21 +136,38 @@ private void cleanUpFinishedOperationsData(Date expirationTime) {
...
+                }
+            }
+            Map<String, List<String>> spaceToFileIds = orhpanedFileEntries.stream()
+                .collect(Collectors.groupingBy(FileEntry::getSpace, Collectors.mapping(FileEntry::getId, Collectors.toList())));
 
-        markOperationsAsCleanedUp(finishedOperations);
+            int removedOldOrhpanedFiles = fileService.deleteAllByFileIds(spaceToFileIds);
+            LOGGER.info("Deleted old orphaned MTA files: " + removedOldOrhpanedFiles);
+        } catch (FileStorageException e) {
+            throw new SLException(e);
```
Comment (enchobelezirev) [inline com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/jobs/CleanUpJob.java]: Maybe some more meaningful message here?

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/jobs/CleanUpJob.java [hunk 3]
```diff
@@ -247,7 +257,11 @@ private void removeProcessLogs(List<String> oldFinishedOperationsIds) {
         return spaceToFileIds;
     }
 
-    private void removeOldFiles(Map<String, List<String>> spaceToFileIds) {
+    private void removeFilesForOperations(List<Operation> operations) {
+        Map<String, String> processIdToAppArchiveId = getProcessIdToAppArchiveId();
```
Comment (enchobelezirev) [inline com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/jobs/CleanUpJob.java]: What would you say for such name: getProcessIdToAppArchiveId -> mapProcessIdToArchive?

Comment (boyan-velinov) [reply com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/jobs/CleanUpJob.java]: I like it => done

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/jobs/CleanUpJob.java (hunk 2 above)
Comment (boyan-velinov) [reply com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/jobs/CleanUpJob.java]: Done

File: com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/jobs/CleanUpJob.java (hunk 1 above)
Comment (boyan-velinov) [reply com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/jobs/CleanUpJob.java]: Done

**Pull request merged**